bridges/primitives/header-chain/tests/implementation_match.rs

```diff
@@ -158,21 +158,28 @@ fn same_result_when_precommit_target_is_not_descendant_of_commit_target() {
 		),
 		Err(Error::PrecommitIsNotCommitDescendant),
 	);
-	// original implementation returns empty GHOST
-	assert_matches!(
-		finality_grandpa::validate_commit(
-			&justification.commit,
-			&full_voter_set(),
-			&AncestryChain::new(&justification.votes_ancestries),
-		)
-		.map(|result| result.ghost().cloned()),
-		Ok(None)
-	);
+
+	// original implementation returns `Ok(validation_result)`
+	// with `validation_result.is_valid() == false`.
+	let result = finality_grandpa::validate_commit(
+		&justification.commit,
+		&full_voter_set(),
+		&AncestryChain::new(&justification.votes_ancestries),
+	)
+	.unwrap();
+
+	assert!(!result.is_valid());
 }
 
 #[test]
 fn same_result_when_justification_contains_duplicate_vote() {
-	let mut justification = make_default_justification(&test_header(1));
+	let mut justification = make_justification_for_header(JustificationGeneratorParams {
```

Comment:
The default justification sets `ancestors: 2` which means that e.g. the precommits finalize block `#3` while the commit says that the target finalized block is `#1`.

Reply:
Thank you! I'll fix it upstream! BTW - we'll probably removed bridges subtree from polkadot repo soon, so you won't need to fix our code anymore.